javascript/atoms/dom.js

```diff
@@ -1136,10 +1136,28 @@ bot.dom.appendVisibleTextLinesFromElement_ = function(elem, lines) {
     // All text nodes that are children of this element need to know the
     // effective "white-space" and "text-transform" styles to properly
     // compute their contribution to visible text. Compute these values once.
-    var whitespace = null, textTransform = null;
+    var whitespace = null, textTransform = null, beforeContentCSS = null, afterContentCSS = null;
     if (shown) {
       whitespace = bot.dom.getEffectiveStyle(elem, 'white-space');
       textTransform = bot.dom.getEffectiveStyle(elem, 'text-transform');
+      beforeContentCSS = goog.style.getComputedStyle(elem, ':before').content;
+      afterContentCSS = goog.style.getComputedStyle(elem, ':after').content;
+      if(!bot.userAgent.IE_DOC_PRE9 && beforeContentCSS.length > 0 && beforeContentCSS!='none'){
+      //trim first and last characters because content puts quotes (single in some browsers and double in others) inside the string.
+        beforeContentCSS=beforeContentCSS.substr(1,beforeContentCSS.length-2);
+        if(lines.length==0)
+          lines.push(beforeContentCSS);
+        else
+          lines[lines.length-1]+=beforeContentCSS;
+      }
+      if(afterContentCSS.length > 0 && afterContentCSS!='none'){
```

Comment:
You should not append the `:after` content until after elem.childNodes have been processed.